### DISCUSSION TITLE
ESP32-S3 Serial Communication using on-board USB-to-UART bridge.

### DISCUSSION THREAD

**Discussion by ImanolGo** [espressif/arduino-esp32#7546] [Q&A] [answered]
Hi, 

I was wondering if it is possible to use the on-board USB-to-UART bridge, for serial communication using, 
Arduino's serial library.

![ESP32-S3-DevKitC-1_v2-SystemBlock](https://user-images.githubusercontent.com/858793/205312853-df891f0b-6884-4d77-8d49-c9555bd2add4.png)

That is, I've created a test board connecting GPIO20 to USB-D+ pin and GPIO19 to USB-D- pin, which should allow USB 2.0 OTG / USB-JTAG bridge.

https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/gpio.html

I would appreciate your help, if you already have experience with this topic.

**Comment by me-no-dev** [comment, marked as answer]
Example: https://github.com/espressif/arduino-esp32/blob/master/libraries/USB/examples/USBSerial/USBSerial.ino
Docs: https://docs.espressif.com/projects/arduino-esp32/en/latest/api/usb_cdc.html
More: https://docs.espressif.com/projects/arduino-esp32/en/latest/tutorials/cdc_dfu_flash.html#usb-cdc

**Reply by ImanolGo** [reply]
Thanks.

- USB CDC On Boot -> Enabled
- Upload Mode -> UART0 / Hardware CDC

Did the trick.
